### DISCUSSION TITLE
Whitelisting wildcards and custom resources in OpenAPI specs

### DISCUSSION THREAD

**Discussion by pakomp** [tailflow/laravel-orion#294] [Bugs]
Hi,

The specification generated by "php artisan orion:specs" seems to use enum for include query param values.

```
public function includes(): array
{
    return ['teachers', 'students'];
}
```

The above will be converted to : 

```
"enum": [
    "teachers",
    "students"
]
```

Issue with wildcards as well. They will be converted to regular strings.
```
public function includes(): array
{
     return ['teachers.*', 'students.*'];
}
```
The above will be converted to : 

```
"enum": [
    "teachers.*",
    "students.*"
]
```

Issue with resources and resource collections as well. 

The following are added in a controller : 
```
 protected $resource = EmployeeResource::class;
 protected $collectionResource = EmployeeCollectionResource::class;
```

**Resources has the following :** 

```
public function toArray($request)
{
      return [];
}

```

The specification still includes resources with properties even if empty array is returned in the actual API.

**Comment by alexzarbn** [comment]
Hi @pakomp,

Thanks for raising the issue. I have noted the mentioned problems, but currently there is no capacity to address them.

Leaving the issue open though. PRs are also welcome :)
